blocks/header/header-megamenu.js

```diff
@@ -18,142 +18,168 @@ export function showRightSubmenu(element) {
-    ).forEach(
-      (el) => el.setAttribute('aria-expanded', 'false'),
-    );
+export async function buildLazyMegaMenus() {
+  const categories = document.querySelectorAll('.menu-nav-category');
+
+  // for each category, get the menu-id attribute
+  categories.forEach(async (category) => {
+    const menuId = category.getAttribute('menu-id');
+
+    await fetch(`/fragments/megamenu/${menuId}.plain.html`, window.location.pathname.endsWith(`/${menuId}`) ? { cache: 'reload' } : {})
+      .then(async (submenuResponse) => {
+        if (submenuResponse.ok) {
+          // eslint-disable-next-line no-await-in-loop
+          const submenuHtml = await submenuResponse.text();
+
+          const submenuContent = div();
+          submenuContent.innerHTML = submenuHtml;
+
+          // get all H2s and create a list of them
+          const h2s = [...submenuContent.querySelectorAll('h2')];
+          const h2List = ul({ class: 'menu-nav-submenu-sections' });
+
+          // add H2s to list
+          h2s.forEach((h2) => {
+            const element = reverseElementLinkTagRelation(h2);
+
+            const h2ListItem = li(
+              { class: 'menu-nav-submenu-section' },
+              element,
+              buildRightSubmenu(element),
+            );
+
+            h2List.append(h2ListItem);
+          });
+
+          // get the list item in the header block that contains a div with attribute menu-id
+          // that matches the menuId
+          const currentMenu = document.querySelector(`h1[id="${menuId}"]`).parentElement.querySelector('.menu-nav-submenu-sections');
+
+          currentMenu.innerHTML = h2List.innerHTML;
+
+          currentMenu.querySelectorAll('.menu-nav-submenu-section').forEach((el) => {
+            el.addEventListener('mouseover', (e) => {
+              e.preventDefault();
+              e.stopPropagation();
+              const rightMenu = e.currentTarget.querySelector('.right-submenu');
+              showRightSubmenu(rightMenu);
+            });
+          });
+        }
+      });
   });
 
-  item.append(closeButton);
-  item.append(submenuContent);
+  const body = document.querySelector('body');
+  body.setAttribute('built-lazy-megamenus', 'true');
 }
 
 export function buildNavbar(content) {
-  const megaMenu = div({ class: 'mainmenu-wrapper sticky-element sticky-desktop' });
-  const container = div({ class: 'container' });
-  const newNav = nav({ id: 'nav' });
-
-  const navTabs = content.querySelector('.nav-menu');
-
-  newNav.innerHTML = navTabs.outerHTML;
-
-  container.append(buildBrandLogo(content));
-  container.append(newNav);
-  megaMenu.append(container);
-
   // link section
   const navMenuUl = ul({ class: 'nav-tabs' });
-  const menus = [...megaMenu.querySelectorAll('.nav-menu > div')];
+  const menus = [...content.querySelectorAll('h1')];
 
   for (let i = 0; i < menus.length; i += 1) {
     const item = li({ class: 'menu-expandable', 'aria-expanded': 'false' });
 
```

Comment:
```suggestion
  for (let i = 0; i < menus.length; i += 1) {
    const item = li({ class: 'menu-expandable', 'aria-expanded': 'false' });

```

code styling; would a for each save some code?

  [...content.querySelectorAll('h1')].forEach((menu) => {
    const text = menu.querySelector('a').textContent
    const item = li({ class: 'menu-expandable', 'aria-expanded': 'false' }, 
      div({ class: 'menu-nav-category', 'menu-id': toClassName(text) }, text,));
    navMenuUl.append(item);
  })